### PR TITLE
Adjust documentation

### PR DESCRIPTION
**Pull request by TimDiekmann**
r? @wencke-lm

### CODE REVIEW

File: docs/slurk_layouts.rst [hunk 1]
```diff
@@ -32,63 +46,116 @@ A layout is a JSON file with a top level dictionary with the following keys:
...
+
+Variables:
+  - ``data.user``: The user who has sent the message
+  - ``data.timestamp``: The timestamp of the message
+  - ``data.private``: A boolean value if this was a direct message or visible to the room
+  - ``data.image``: The image URL
+  - ``data.width``: The width of the sent image
+  - ``data.height``: The height of the sent image
+Examples:
+  - ``"display-image"``: Displays ss a simple image
```
Comment (wencke-lm) [inline docs/slurk_layouts.rst:103]:
```suggestion
  - ``"display-image"``: Displays a simple image
```
🐍

File: docs/slurk_layouts.rst [hunk 2]
```diff
@@ -32,63 +46,116 @@ A layout is a JSON file with a top level dictionary with the following keys:
...
+  - ``"markdown"``: Prints as formatted markdown if tagged as html
+
+``"incoming-image"``
+--------------------
+Displays images in the chat area they arrive
+
+Variables:
+  - ``data.user``: The user who has sent the message
+  - ``data.timestamp``: The timestamp of the message
+  - ``data.private``: A boolean value if this was a direct message or visible to the room
```
Comment (wencke-lm) [inline docs/slurk_layouts.rst:98]:
```suggestion
  - ``data.private``: A boolean value indicating whether this was a direct message or visible to the room
```

File: docs/slurk_layouts.rst [hunk 3]
```diff
@@ -32,63 +46,116 @@ A layout is a JSON file with a top level dictionary with the following keys:
...
+  - ``data.message``: The message string of the sent message if any
+  - ``data.html``: Tag if the message is marked as HTML
+Examples:
+  - ``"display-text"``: Prints as plain text
+  - ``"markdown"``: Prints as formatted markdown if tagged as html
+
+``"incoming-image"``
+--------------------
+Displays images in the chat area they arrive
```
Comment (wencke-lm) [inline docs/slurk_layouts.rst:93]:
```suggestion
Displays images in the chat area as they arrive
```

File: docs/slurk_gettingstarted.rst [hunk 4]
```diff
@@ -104,40 +101,49 @@ Chatting with a bot
 ~~~~~~~~~~~~~~~~~~~
 
 Without additional environment variables, the server uses an in-memory
-database and resets on every restart of the server. Now let's restart
-the server to reset the database:
+database and resets on every restart of the server. If you restarted the
```
Comment (wencke-lm) [inline docs/slurk_gettingstarted.rst:104]:
```suggestion
database and resets on every restart of the server. If you restarted
```

**Review by wencke-lm — COMMENTED** [review]
Mostly spelling, only a few comments on the content.

File: README.md [hunk 5]
```diff
@@ -6,67 +6,43 @@
...
+Now you can open `localhost` in your browser and log in with any username and the provided admin token
 
-```
-pip install sphinx sphinx_rtd_theme
-cd docs
-make html
-```
 
-The full documentation can then be found at *docs/_build/*
+The [full documentation][doc] can be found on GitHUb.
```
[the remark this answers is not in the document]
Comment (TimDiekmann) [reply README.md:38]: The documentation is linked to the built documentation with `[full documentation][doc]`.

File: docs/slurk_api.rst [hunk 6]
```diff
@@ -7,3 +7,49 @@ REST API for slurk
 =================================================
 
 `View in RapiDoc <slurk_api_rapidoc.html>`_
+
+slurk uses a Restful API to modify the underlying database. In the link above, the full documentation
```
Comment (janagoetze) [inline docs/slurk_api.rst:11]: RESTful

File: docs/slurk_api.rst [hunk 7]
```diff
@@ -7,3 +7,49 @@ REST API for slurk
...
+for each method is shown. When the slurk server is running, you can also visit ``/rapidoc``, ``/redoc``,
+or ``/swagger-ui`` to see the API and eventually try it out directly.
```
Comment (janagoetze) [inline docs/slurk_api.rst:13]: ~eventually~

File: docs/slurk_api.rst [hunk 8]
```diff
@@ -7,3 +7,49 @@ REST API for slurk
...
+
+The API is more or less sorted in chronical order, as many tables requires a previously returned
```
[the remark this answers is not in the document]
Comment (janagoetze) [reply docs/slurk_api.rst:15]: chronical -> chronological

File: docs/slurk_api.rst [hunk 9]
```diff
@@ -7,3 +7,49 @@ REST API for slurk
...
+-------------------------
+
+A Layout defines how a room looks like. A room could either be a room, which is created by hand
+or it could be created from a task. For more information on layouts have a look at :ref:`slurk_layouts`.
+
+Users
+-----
+
+A user is either a human user, mostly with very limited permissions, or a bot. Every user needs a
+token to login. While a token can be used to register multiple users, it's not possible to have multiple
```
Comment (janagoetze) [inline docs/slurk_api.rst:28]: to log in

File: docs/slurk_api.rst [hunk 10]
```diff
@@ -7,3 +7,49 @@ REST API for slurk
...
+tokens as user. Security and permissions are important aspects to consider when giving access to
+bots and users. A token is associated with a permissions list and should only contain permissions,
+which are really needed for the corresponding user (human or bot). Especially, the permissions to send
+html messages have to be avoided when giving out to human users, as they send inject any html objects.
+
+Also, the ``"api"``-permissions is critical to give to unknown users, as with the api-token, everything
```
[the remark this answers is not in the document]
Comment (janagoetze) [reply docs/slurk_api.rst:34]: suggestion: "should not be given to arbitrary or unknown users"

File: docs/slurk_deployment.rst [hunk 11]
```diff
@@ -4,236 +4,109 @@
...
-`this tutorial <https://www.digitalocean.com/community/tutorials/how-to-serve-flask-applications-with-gunicorn-and-nginx-on-ubuntu-18-04>`_.
-
-First, isolate your Flask application on your computer. For this you have to install *virtualenv* package.
+slurk is backed by a database. When no database is provided, an in-memory database is used.
+This can be changed with setting ``SLURK_DATABASE_URI``. For a list of possible engines see
+`the SQLAlchemy documentation <https://docs.sqlalchemy.org/en/14/core/engines.html#database-urls>`_,
+however only SQLite and Postgres are tested. If you notice errors with other providers, please
+feel free to `file an issue <https://github.com/clp-research/slurk/issues/new>`_. Also note, that
+SQLite behaves weirdly when using auto-increment keys, so deletion of entries may result in the
+same id being used again. Later, we will see, how slurk can be set up with a Postgres-database and
```
Comment (janagoetze) [inline docs/slurk_deployment.rst:23]: let's kill the commas

File: docs/slurk_deployment.rst [hunk 12]
```diff
@@ -4,236 +4,109 @@
...
-  nano ~/server/wsgi.py
+For spinning up an OpenVidu server, please consult the `corresponding documentation <https://docs.openvidu.io/en/2.18.0/deployment/>`_.
 
-The chat instance itself is created in ``~/server/app/__init__.py``, and that is why we will use this instance in our `wsgi.py` file.
-Import this instance into the file, save and close it once you are finished
+Hosting slurk
+~~~~~~~~~~~~~
 
-.. code-block:: python
+There is a script in the slurk-repository called ``scripts/start_server.sh``. depending
```
Comment (janagoetze) [inline docs/slurk_deployment.rst:44]: Depending

File: docs/slurk_deployment.rst [hunk 13]
```diff
@@ -4,236 +4,109 @@
...
+on the environment variables set, the scripts starts a GUnicorn server locally or in a
```
Comment (janagoetze) [inline docs/slurk_deployment.rst:45]: script

File: docs/slurk_deployment.rst [hunk 14]
```diff
@@ -4,236 +4,109 @@
...
+See the `Deployment Options <https://flask.palletsprojects.com/en/2.0.x/deploying/>`_ of
+Flask for a big list of different options.
+
+
+Example using docker-compose and Postgres
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Before spinning up slurk, we need a proper database. For our case, we chose postgres.
+When using the postgres docker container and the slurk docker container, the two
+container probably can't see each other, so we use `docker-compose <https://docs.docker.com/compose/>`_ here.
```
Comment (janagoetze) [inline docs/slurk_deployment.rst:80]: containers

File: docs/slurk_layouts.rst (hunk 1 above)
Comment (janagoetze) [reply docs/slurk_layouts.rst:103]: 🐍

File: docs/slurk_layouts.rst [hunk 15]
```diff
@@ -32,63 +46,116 @@ A layout is a JSON file with a top level dictionary with the following keys:
...
+  - ``"plain-history"``: As plain text and images
+  - ``"markdown-history"``: Formatted as markdown if tagged as html
+  - ``"attribute-history"``: Applies previous changes to the layout
+
+``"typing-users"``
+------------------
+Called when state of currently typing users is changed
+
+Variables:
+  - ``users``: A map of currently typing users, with its id as the key
```
Comment (janagoetze) [inline docs/slurk_layouts.rst:142]: with whose id as the key?

**Review by janagoetze — COMMENTED** [review]
impressive docs! again, I volunteer to take care of typos and related boring stuff

File: README.md [hunk 16]
```diff
@@ -6,67 +6,43 @@
...
+[2021-07-24 10:49:41 +0000] [8] [INFO] Booting worker with pid: 8
+[2021-07-24 10:49:42,779] WARNING in database: SQLite should not be used in production
+admin token:
+01234567-89ab-cdef-0123-456789abcdef
 ```
 
-5. open a browser tab on `localhost:5000` and login with any username and the admin token
-
-If you want to build the documentation yourself, you need the packages _sphinx_ and _sphinx_rtd_theme_. Then you can create the documentation in the _docs_ folder:
+Now you can open `localhost` in your browser and log in with any username and the provided admin token
```
[the remark this answers is not in the document]
Comment (TimDiekmann) [reply README.md:35]: You can, but you can't send any message. 🙂

[the remark this answers is not in the document]
Comment (wencke-lm) [reply README.md]: Screenshot (158)

[the remark this answers is not in the document]
Comment (TimDiekmann) [reply README.md]: Ah, right, the admin token does not have a room associated with it...

File: docs/slurk_about.rst [hunk 17]
```diff
@@ -45,7 +45,7 @@ Technical concepts
   in.
 
   **Commands**: is basically the same as a public text message, but is only sent to the users with the required
-  permissions.
+  permissi
```
Comment (wencke-lm) [inline docs/slurk_about.rst:48]: what happened here?

File: docs/slurk_api.rst [hunk 18]
```diff
@@ -8,11 +8,11 @@ REST API for slurk
...
 `View in RapiDoc <slurk_api_rapidoc.html>`_
 
-slurk uses a Restful API to modify the underlying database. In the link above, the full documentation
+slurk uses a RESTful API to modify the underlying database. In the link above, the full documentation
 for each method is shown. When the slurk server is running, you can also visit ``/rapidoc``, ``/redoc``,
-or ``/swagger-ui`` to see the API and eventually try it out directly.
+or ``/swagger-ui`` to see the API and try it out directly.
 
-The API is more or less sorted in chronical order, as many tables requires a previously returned
+The API is more or less sorted in chronological order, as many tables reqire a previously returned
```
Comment (wencke-lm) [inline docs/slurk_api.rst:15]:
```suggestion
The API is more or less sorted in chronological order, as many tables require a previously returned
```
almost^^

File: docs/slurk_api.rst [hunk 19]
```diff
@@ -25,13 +25,13 @@ Users
 -----
 
 A user is either a human user, mostly with very limited permissions, or a bot. Every user needs a
-token to login. While a token can be used to register multiple users, it's not possible to have multiple
-tokens as user. Security and permissions are important aspects to consider when giving access to
+token to log in. While a token can be used to register multiple users, it's not possible assign more
```
Comment (wencke-lm) [inline docs/slurk_api.rst:28]:
```suggestion
token to log in. While a token can be used to register multiple users, it's not possible to assign more
```

File: docs/slurk_deployment.rst [hunk 20]
```diff
@@ -11,16 +11,16 @@ slurk reads a few environment variables for configuration. Because slurk is a fl
...
 slurk is backed by a database. When no database is provided, an in-memory database is used.
-This can be changed with setting ``SLURK_DATABASE_URI``. For a list of possible engines see
+This can be changed by setting ``SLURK_DATABASE_URI``. For a list of possible engines, see
 `the SQLAlchemy documentation <https://docs.sqlalchemy.org/en/14/core/engines.html#database-urls>`_,
 however only SQLite and Postgres are tested. If you notice errors with other providers, please
-feel free to `file an issue <https://github.com/clp-research/slurk/issues/new>`_. Also note, that
-SQLite behaves weirdly when using auto-increment keys, so deletion of entries may result in the
-same id being used again. Later, we will see, how slurk can be set up with a Postgres-database and
+`file an issue <https://github.com/clp-research/slurk/issues/new>`_. Also note, that
+SQLite behaves weirdly when using auto-increment keys, so deletion of entries may cause the
```
Comment (wencke-lm) [inline docs/slurk_deployment.rst:22]: I liked the earlier version more 🤔 or ``may cause the same id to be used again``

File: docs/slurk_layouts.rst [hunk 21]
```diff
@@ -139,7 +139,7 @@ Examples:
 Called when state of currently typing users is changed
 
 Variables:
-  - ``users``: A map of currently typing users, with its id as the key
+  - ``users``: A map of currently typing users, with whose id as the key
```
Comment (wencke-lm) [inline docs/slurk_layouts.rst:142]: I think jana really did not understand who was meant and didn't intend to replace its by whose.
```suggestion
  - ``users``: A map of currently typing users, with their IDs as keys
```

File: docs/slurk_about.rst (hunk 17 above)
Comment (TimDiekmann) [reply docs/slurk_about.rst:48]: No clue...

**Review by wencke-lm — APPROVED** [review]

**Pull request merged**